Highway.cpp

```diff
@@ -13,34 +15,54 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
+    if(Motorcycle* m = dynamic_cast<Motorcycle*>(v))
+    {
+        m->setSpeed(speedLimit);
+    }
+    if(Car* c = dynamic_cast<Car*>(v))
+    {
+        c->setSpeed(speedLimit);
+    }
+    if(SemiTruck* t = dynamic_cast<SemiTruck*>(v))
+    {
+        t->setSpeed(speedLimit);
+    }
 }
 
 void Highway::removeVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that tries to evade the cops. 
-
     trucks pull over, but cars and bikes try to evade!!
     */
+    if(Motorcycle* m = dynamic_cast<Motorcycle*>(v))
+    {
+        m->tryToEvade();
+    }
+    if(Car* c = dynamic_cast<Car*>(v))
+    {
+        c->tryToEvade();
+    }
+    //trucks dont try to evade
+    
+    
```

Comment:
Trucks pull over.
Call the Truck function that pulls over
also, use `else if` and `auto*`
- [ ] revise